### PR TITLE
Feat: add polarizability fitting net

### PR DESCRIPTION
**Pull request by anyangml**
This PR is to provide pytorch implementation and backend-independent numpy implementation of the polarizability fitting net. 

Note:
- The shift_diag requires statistics, not implemented in this PR.

### CODE REVIEW

**Comment by wanghan-iapcm** [comment]
#3289 will be merged before this PR.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: deepmd/pt/model/task/fitting.py [hunk 1]
```diff
@@ -585,15 +589,23 @@ def _forward_common(
                 atom_property = (
                     self.filter_layers.networks[0](xx) + self.bias_atom_e[atype]
                 )
-                outs = outs + atom_property  # Shape is [nframes, natoms[0], 1]
+                outs = (
+                    outs + atom_property
+                )  # Shape is [nframes, natoms[0], net_dim_out]
+                if hasattr(self, "scale"):
```
[the remark this answers is not in the document]
Comment (anyangml) [reply deepmd/pt/model/task/fitting.py:595]: the alternative would be pass those parameters into the parent class. However, these parameters are specific to polar fitting. We would have to define those parameters out side of the super()__init__() in sibling classes.

[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/pt/model/task/fitting.py:595]: I don't see the necessity to multiply scale in the parent class.

[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/pt/model/task/fitting.py:595]: The multiplication can be implemented in the child class.

[the remark this answers is not in the document]
Comment (anyangml) [reply deepmd/pt/model/task/fitting.py:595]: I think leave it in the parent class might be better, since `dipole` may also use it.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: deepmd/dpmodel/fitting/polarizability_fitting.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            gr is not None
+        ), "Must provide the rotation matrix for polarizability fitting."
+        # (nframes, nloc, m1)
+        out = self._call_common(descriptor, atype, gr, g2, h2, fparam, aparam)[
+            self.var_name
+        ]
+        out = out * self.scale[atype]
+        if self.fit_diag:
+            out = out.reshape(-1, self.embedding_width)  # (nframes * nloc, m1)
+            out = np.stack([np.diag(v) for v in out])  # (nframes * nloc, m1, m1)
```
[the remark this answers is not in the document]
Comment (anyangml) [reply deepmd/dpmodel/fitting/polarizability_fitting.py]: please take a closer look at the new implementation. It seems that it's not consistent with the old one.

**Review by njzjz — APPROVED** [review]

File: deepmd/dpmodel/fitting/polarizability_fitting.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply deepmd/dpmodel/fitting/polarizability_fitting.py]: Could you plz explain why they are not consistent?

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**